src/oop/object_interactions/medium/q6/MainTest.java

```diff
@@ -0,0 +1,197 @@
+                    "You have modified the provided " + setNumPassengersMethodName + "method  on the " + className + " class. Please revert it to the original state.");
+            assertNull(output, String.join(" ", "Your", setNumPassengersMethodName, "should not return any output"));
+        }
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(20, 10, 40, 35),
+                Arguments.of(10, 20, 15, 30),
+                Arguments.of(150, 200, 100, 50),
+                Arguments.of(20, 20, 10, 20)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctTransferPassengersMethod(int seat, int pass, int busySeat, int busyPass) throws Throwable {
+        Object[][] arguments = {
+                {seat, int.class},
+                {pass, int.class}
+        };
+        Airplane busy = new Airplane(busySeat, busyPass);
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] methodArguments = {
+                {busy, Airplane.class}
+        };
+        Object output = null;
+        if (seat - pass > 0) {
+            int move = Math.min(busyPass - busySeat, seat - pass);
+            output = testClass.callMethod(transferPassengersMethodName, methodArguments, new String[]{"public"}, classInstance, new Clause[]{
+                    new StringLiteral("We moved " + move + " passengers"),
+                    new Optional(new StringLiteral(" "))
+            });
+            _assertEquals(pass + move, testClass.getFieldValue(classInstance, numPassengersAttributeName),
+                    "Your " + transferPassengersMethodName + " method does not correctly add the " + numPassengersAttributeName + " attribute of the recipient.");
+            _assertEquals(busyPass - move, busy.getNumPassengers(),
+                    "Your " + transferPassengersMethodName + " method does not correctly decrease the " + numPassengersAttributeName + " attribute of the donor.");
+        } else {
+            output = testClass.callMethod(transferPassengersMethodName, methodArguments, new String[]{"public"}, classInstance, new Clause[]{
+                    new StringLiteral("This plane is full too and cannot accommodate anymore passengers"),
+                    new Optional(new StringLiteral(" "))
+            });
+            _assertEquals(pass, testClass.getFieldValue(classInstance, numPassengersAttributeName),
+                    "Your " + transferPassengersMethodName + " method does not correctly add the " + numPassengersAttributeName + " attribute of the recipient.");
+            _assertEquals(busyPass, busy.getNumPassengers(),
+                    "Your " + transferPassengersMethodName + " method does not correctly decrease the " + numPassengersAttributeName + " attribute of the donor.");
+            _assertEquals(seat, testClass.getFieldValue(classInstance, maxSeatsAttributeName),
```

Comment:
I do not understand the purpose of the two maxSeats asserts, from what I understand the method does not alter the value of that attribute, and if it did it would not make sense to, since the number of total seats on an airplane doesn't generally change